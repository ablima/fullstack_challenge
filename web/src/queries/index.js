import gql from "graphql-tag";

export const GET_PRODUCTS = gql`
  query Products($id: Int, $name: String, $categoryId: Int){
    getProducts(id: $id, name: $name, categoryId: $categoryId){
      id
      name
      description
      qnt
      price
      thumbnails
    }
  }
`

export const ADD_PRODUCT_TO_CART = gql`
  mutation Product($id: Int){
    addProductToCart(id: $id){
      id
    }
  }
`;

export const REMOVE_PRODUCT_FROM_CART = gql`
  mutation Product($id: Int){
    removeProductFromCart(id: $id){
      id
    }
  }
`;

export const GET_CATEGORIES = gql`
  query {
    getCategories{
      id
      name
    }
  }
`;

export const CREATE_ORDER = gql`  
  mutation Order($order: NewOrder){
    createOrder(order: $order){
      id
    }
  }
`;