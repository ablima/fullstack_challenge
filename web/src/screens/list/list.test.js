import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router';
import { MockedProvider } from '@apollo/client/testing';
import { GET_PRODUCTS } from '../../queries';
import ListScreen from './index';

const mocks = [
  {
    request: {
      query: GET_PRODUCTS
    },
    result: {
      data: {
        getProducts: [
          {
            name: "Produto",
            description: {},
            qnt: 5,
            price: 100,
            categoryId: 1,
            thumbnails: "images/celular1-1.jpg"
          },
          {
            name: "Produto 2",
            description: {},
            qnt: 5,
            price: 100,
            categoryId: 2,
            thumbnails: "images/celular1-1.jpg;images/celular1-2.jpg"
          }
        ]
      }
    }
  }
];

test('Check if render is correct', () => {
  const component = renderer.create(
    <MockedProvider mocks={mocks} addTypename={false}>
      <MemoryRouter initialEntries={["/1"]}>
        <ListScreen />
      </MemoryRouter>      
    </MockedProvider>    
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});