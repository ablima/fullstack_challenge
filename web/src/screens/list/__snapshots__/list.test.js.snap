// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if render is correct 1`] = `
<div
  id="outer-container"
>
  <div
    className="header"
    id="header"
  >
    <img
      className="logo"
      onClick={[Function]}
      src="storeLogo.png"
    />
    <div
      className="cartIcon"
    >
      <img
        className="icon"
        onClick={[Function]}
        src="cart.svg"
      />
      <div
        className="animated"
        style={
          Object {
            "animationDelay": "undefinedms",
            "animationDuration": "undefinedms",
            "opacity": 0,
            "pointerEvents": "none",
          }
        }
      >
        <div
          className="cartContainer"
        >
          <span>
            Sem produtos no carrinho
          </span>
        </div>
      </div>
    </div>
  </div>
  <main
    id="page-wrap"
  >
    <div
      className="mainContent"
    >
      <div
        className="categoriesContainer"
      />
      <div
        className="cardContainer"
      />
    </div>
  </main>
</div>
`;
